### ISSUE TITLE
Read JObject Property: System.Security.VerificationException: Operation could destabilize the runtime

### ISSUE THREAD

**Issue by kelilive** [JamesNK/Newtonsoft.Json#2915]
In order to avoid duplication, I made a quote. A large part of the problem lies with Newtonsoft itself.
Please see: https://github.com/dotnet/runtime/issues/95445

**Comment by kelilive** [comment]
```csharp
RawModbusRtuOverTcp.Test1();

Console.WriteLine("--------");
RawModbusRtuOverTcp.Test2(1, 1);

Console.WriteLine("--------");
RawModbusRtuOverTcp.Test3(1, 1);

Console.WriteLine("--------");
RawModbusRtuOverTcp.Test4(1, 1);
```

I tried a more primitive approach, but I don't know how to parse this data that returns normal int, float, double. The following is a description of the problems I have encountered. I hope some warm-hearted friends can give me some help.

When I run it for the first time, Test1,Test2, and Test3 can get the data normally, and reading a simple length of data is the same result. However, if Test4 cannot be obtained, it will be stuck at the time of Receive and cannot proceed to the next line. I have slightly improved the code. I don't know what the problem is. Now only the first one can return the result, and the other 3 can't return the data.

[RawModbusRtuOverTcp.zip](https://github.com/JamesNK/Newtonsoft.Json/files/13741940/RawModbusRtuOverTcp.zip)
